compiler-rt/lib/tysan/tysan_interceptors.cpp

```diff
@@ -108,6 +108,14 @@ INTERCEPTOR(void *, malloc, uptr size) {
   return res;
 }
 
+#if SANITIZER_APPLE
+INTERCEPTOR(uptr, malloc_size, void *ptr) {
+  if (DlsymAlloc::Use())
```

Comment:
A less fragile implementation here would be ask the `DlsymAlloc` allocator if it owns this allocation and if so request the size from it, otherwise call the system `malloc_size`. That way `malloc_size` always reports the right things no matter which allocator actually allocated the memory.

The current implementation means that if `malloc_size` is called **after** TYSan init on an allocation that was allocated during TySan init then we'll call the system `malloc_size` which will fail because the system allocator didn't allocator the memory.

Reply:
Thanks, I think that should be taken care of by checking `PointerIsMine` (as mentioned by @fmayer ). Update the code.